types/src/block_info.rs

```diff
@@ -139,7 +139,8 @@ impl BlockInfo {
 
     /// The epoch after this block committed
     pub fn next_block_epoch(&self) -> u64 {
-        self.next_epoch_state().map_or(self.epoch(), |e| e.epoch)
+        self.next_epoch_state()
```

Comment:
original was probably cheaper?

Reply:
it should be a function call vs branch not taken?

Reply:
function call vs creating an object for a function pointer, that captures "self" (for `|| self.epoch`).

Reply:
I wonder if the compiler can statically map the function call, but with closures it likely works like you described. Still, the docs seem to suggest using || with functions, and it makes sense to me as a pattern, since the function may easily change, etc. If you don't disagree, I'll keep that as a convention for now.

Reply:
oh but here there is no call even, fixing.